app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/extension/ExtensionHandler.java

```diff
@@ -247,7 +250,26 @@ public void install(@NotNull @PathParam("id") final String id) {
     }
 
     @Override
-    public ListResult<Extension> list(UriInfo uriInfo) {
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            name = "sort", value = "Sort the result list according to the given field value",
+            paramType = "query", dataType = "string"),
+        @ApiImplicitParam(
+            name = "direction", value = "Sorting direction when a 'sort' field is provided. Can be 'asc' " +
+            "(ascending) or 'desc' (descending)", paramType = "query", dataType = "string"),
+        @ApiImplicitParam(
+            name = "page", value = "Page number to return", paramType = "query", dataType = "integer", defaultValue = "1"),
+        @ApiImplicitParam(
+            name = "per_page", value = "Number of records per page", paramType = "query", dataType = "integer", defaultValue = "20"),
+        @ApiImplicitParam(
+            name = "query", value = "The search query to filter results on", paramType = "query", dataType = "string"),
+        @ApiImplicitParam(
+            name = "extensionType", value = "The type to filter results on", paramType = "query", dataType = "string"),
+    })
+    public ListResult<Extension> list(@Context UriInfo uriInfo) {
+        String extensionType = uriInfo.getQueryParameters().getFirst("extensionType");
```

Comment:
```suggestion
    })
    public ListResult<Extension> list(@Context UriInfo uriInfo, @QueryParam("extensionType") @DefaultValue("Steps") @ApiParam(required = false) Extension.Type extensionType) {
```

Reply:
Thanks @zregvart. Please, notice that the method still overrides from Lister. Your suggestion is what I'd like to do as an additional task once we reach consensus on [this thread](https://groups.google.com/forum/#!topic/syndesis/PGJ1A6meKoQ).

Reply:
Lister is not a contract, Lister is a mixin. Make it not override from Lister, if you look at the API the end result is the same. If you look at the code this option is much better. Though I think we should not use reflection for filtering this suggestion has no relevance toward reflection. This suggestion is about making the API contract explicit and type safe.

Reply:
I've applied the suggestion. I wanted to work separately on this feature and later on cleaning everything that was depending by that List interface. As that was born by a discussion with you, please feel free to chime in the thread: https://groups.google.com/forum/#!topic/syndesis/PGJ1A6meKoQ

Reply:
Actually, you did not apply the suggestion.

Reply:
Hey @zregvart. If I read correctly you proposed a code change and  "Make it not override from Lister". This was provided by this commit: 34bd4d09cdbb1abb72f4a9030f3d95782fc179e7. Were you suggesting something else?

Reply:
Did you notice the suggestion diff above?

Reply:
I've added another commit (34bd4d09cdbb1abb72f4a9030f3d95782fc179e7) with the change requested, [this is the final result](https://github.com/syndesisio/syndesis/blob/master/app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/extension/ExtensionHandler.java#L268). A part the default value, the rest is the same. Or maybe your're talking about something else?

Reply:
@squakez if you carefully examine the suggestion diff displayed above this conversation, you might notice that the extension type parameter is of `Extension.Type` and not fetched from from `uriInfo.getQueryParameters()`

Reply:
Thanks, I see that. I've provided a `String` parameter instead. I blindly assumed it had to be a String and to be converted. I'll log a new issue to fix that. The thread I've opened is still idle, I have bandwidth to work on a general improvement if I got some positive feedback though.